lock-api/src/main/java/com/palantir/lock/BareLockRpcClient.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * (c) Copyright 2019 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.lock;
+
+import java.math.BigInteger;
+import java.util.Optional;
+import java.util.Set;
+
+import javax.ws.rs.Consumes;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+
+import com.palantir.annotations.remoting.CancelableServerCall;
+import com.palantir.logsafe.Safe;
+
+/**
+ * See also {@link LockRpcClient}. This interface exists to support legacy users not using an external TimeLock
+ * server (as in these cases, lock services will not be namespaced) but also using remoting frameworks where
+ * endpoints producing null values are not permitted.
+ */
+@Path("/lock")
+@Produces(MediaType.APPLICATION_JSON)
+@Consumes(MediaType.APPLICATION_JSON)
+public interface BareLockRpcClient {
```

Comment:
from previous discussion, perhaps `NamespaceAgnosticLockRpcClient`, which captures the intent of leader lock rpc client not needing a namespace and the timelock namespace agnostic adaptor.

Reply:
Yep, will change to that